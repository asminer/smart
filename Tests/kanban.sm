


/*

  The kanban manufacturing system example

*/

pn k(int n) := {

  place pm1, pback1, pkan1, pout1,
        pm2, pback2, pkan2, pout2,
        pm3, pback3, pkan3, pout3,
        pm4, pback4, pkan4, pout4;

  trans tin1, tredo1, tok1, tback1,
        tsynch1_23,
	tredo2, tok2, tback2, 
        tredo3, tok3, tback3,
	tsynch23_4,
        tredo4, tok4, tback4, tout4;
 
  arcs(pkan1:tin1, tin1:pm1, pm1:tredo1, pm1:tok1, tredo1:pback1, tok1:pout1, 
       pback1:tback1, tback1:pm1, pout1:tsynch1_23, tsynch1_23:pkan1,

       pkan2:tsynch1_23, tsynch1_23:pm2, pm2:tredo2, pm2:tok2, tredo2:pback2,
       tok2:pout2, pback2:tback2, tback2:pm2, pout2:tsynch23_4, 
       tsynch23_4:pkan2,

       pkan3:tsynch1_23, tsynch1_23:pm3, pm3:tredo3, pm3:tok3, tredo3:pback3,
       tok3:pout3, pback3:tback3, tback3:pm3, pout3:tsynch23_4, 
       tsynch23_4:pkan3,

       pkan4:tsynch23_4, tsynch23_4:pm4, pm4:tredo4, pm4:tok4, tredo4:pback4,
       tok4:pout4, pback4:tback4, tback4:pm4, pout4:tout4, tout4:pkan4
       );

  init(pkan1:n, pkan2:n, pkan3:n, pkan4:n);
  
  bigint m1 := num_states;
  bigint m2 := num_arcs;
};

for (int i in {1..3}) {
  print("Kanban model with N=", i, "\n\f");
  compute(k(i).m2);
  print("\tstates: ", k(i).m1, "\n");
  print("\t edges: ", k(i).m2, "\n\f");
}

