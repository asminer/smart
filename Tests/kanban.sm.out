Kanban model with N=1
	states: 160
	 edges: 616
Kanban model with N=2
	states: 4600
	 edges: 28120
Kanban model with N=3
	states: 58400
	 edges: 446400
